heron/tools/tracker/src/python/tracker.py

```diff
@@ -64,17 +66,18 @@ def sync_topologies(self) -> None:
     def on_topologies_watch(state_manager: StateManager, topologies: List[str]) -> None:
       """watch topologies"""
       topologies = set(topologies)
-      Log.info("State watch triggered for topologies.")
-      Log.debug("Topologies: %s", topologies)
-      cached_names = {t.name for t in self.get_stmgr_topologies(state_manager.name)}
-      Log.debug("Existing topologies: %s", cached_names)
-      for name in cached_names - topologies:
-        Log.info("Removing topology: %s in rootpath: %s",
-                 name, state_manager.rootpath)
-        self.remove_topology(name, state_manager.name)
-
-      for name in topologies - cached_names:
-        self.add_new_topology(state_manager, name)
+      Log.info(f"State watch triggered for topologies of {state_manager.name}.")
+      Log.debug(f"Received topologies: {state_manager.name}, {topologies}")
+      cached_names = [t.name for t in self.get_stmgr_topologies(state_manager.name)]
```

Comment:
was there a reason to convert this from a set to a list? the code below still ends up doing what are essentially set operations, just implicitly and slower

Reply:
Updated